pkg/driver/controller_server.go

```diff
@@ -930,7 +930,10 @@ func (driver *Driver) controllerUnpublishVolume(volumeID string, nodeID string,
 	storageProvider, err := driver.GetStorageProvider(secrets)
 	if err != nil {
 		log.Error("err: ", err.Error())
-		return status.Error(codes.Internal,
```

Comment:
does this apply to ControllerPublish as well? I see we use `Internal` error code with multiple failures.

Reply:
Ya it does https://github.com/kubernetes-csi/external-attacher/blob/ad7d376eb4ac985860e37903590ce6a17b340b06/pkg/attacher/attacher.go#L72 
but I looked at those error codes and those are actual volume creation errors and we shouldn't retry. I'll double check if we missed any

Reply:
creation? this is during ControllerPublish right? I was checking calls like GetStorageProvider(), GetNodeContext() and SetNodeContext() which returns `Internal` error code. May be on certain http status codes we should indicate caller to derive at error code which can be retried or always return `Unavailable`, @george?

Reply:
Ya, I looked at volume creation ones. GetStorageProvider(), GetNodeContext() and SetNodeContext() could be retried.